qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -147,7 +147,8 @@ def compute(
         self._compute()
 
     def _compute(self) -> None:
-        self._parameters = np.zeros(len(self._excitation_list))
+        reps = self._ansatz.reps if self._ansatz is not None else 1
```

Comment:
When do you expect `self._ansatz` to be `None`? Because in such a scenario this could still not work as expected, right?

Reply:
Because one can bypass the ansatz and set the excitation list directly in these old classes. I think this would cover such case. In the new classes this is no longer possible.